vbml/patcher/abc.py

```diff
@@ -39,17 +39,19 @@ def validator(
         """
 
         def decorator(validator_handler: ValidatorType) -> ABCValidator:
-            validator: ABCValidator
+            validator: typing.Optional[ABCValidator] = None
 
-            if isinstance(validator_handler, types.FunctionType):
+            if inspect.isclass(validator_handler):
+                if issubclass(validator_handler, ABCValidator):  # type: ignore
+                    if key is not None:
+                        validator_handler.key = key
+                    validator = validator_handler()  # type: ignore
+            elif callable(validator_handler):
                 validator = FuncBasedValidator(
                     key or validator_handler.__name__, validator_handler
                 )
-            elif issubclass(validator_handler, ABCValidator):  # type: ignore
-                if key is not None:
-                    validator_handler.key = key
-                validator = validator_handler()  # type: ignore
-            else:
+
+            if validator is None:
```

Comment:
why not `else`? what if not isclass and not callable

Reply:
In that case `validator` will be `None`. That's actually do the same thing as before.

Reply:
I see, you fill the variable